### PR TITLE
check if metadataReader is not null - thos is solving problem with fa…

### PR DESCRIPTION
**Pull request by Sasinkas**
…iling ERR998/7

I had to redo this PR (https://github.com/microsoft/binskim/pull/1097) due to a failed rebase.
The most essential change here ensures that metadataReader doesn't fail when encountering a null value. Once this assertion is in place, the rest of the logic remains stable.

### CODE REVIEW

File: src/BinaryParsers/PEBinary/PortableExecutable/PE.cs [hunk 1]
```diff
@@ -527,7 +527,7 @@ public bool IsManagedResourceOnly
                     return this.isManagedResourceOnly.Value;
                 }
 
-                this.isManagedResourceOnly = this.metadataReader.MethodDefinitions.Count == 0;
+                this.isManagedResourceOnly = this.metadataReader?.MethodDefinitions.Count == 0;
```
[the remark this answers is not in the document]
Comment (Sasinkas) [reply src/BinaryParsers/PEBinary/PortableExecutable/PE.cs:530]: isManaged does not determine whether metadataReader is null. It's possible for isManaged to be false while metadataReader is also null.

**Review by martin-reznik — APPROVED** [review]

File: src/BinaryParsers/PEBinary/PortableExecutable/PE.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelcfanning) [reply src/BinaryParsers/PEBinary/PortableExecutable/PE.cs:530]: I think you mean it's possible for IsManaged to be true while the metadata reader is null. It's pretty clear from the code that this shouldn't happen except under very adverse conditions, i.e., one of the three exception kinds below. Are you absolutely sure that when the code paths below are hit, that we will continue to execute and call `IsManagedResourceOnly`?

Should we actually be considering that the IsManaged check should be extended to include a test for a non-null metadata reader? 

I don't like what you've done here because you appear to have spot-fixed one code path only, the crash that you happened to hit. If you believe that we should continue to attempt analysis when the metadataReader is null, why aren't we scrubbing for this generally across the code?

Even in your code change below, if metadata reader is null in the place you put the protection, the very *next* line looks like it might crash.

I apologize but I just don't agree with your approach. You don't need to be blocked just because I disagree, however, you can have Martin approve it. 

            catch (IOException e) { this.LoadException = e; }
            catch (BadImageFormatException e) { this.LoadException = e; }
            catch (UnauthorizedAccessException e) { this.LoadException = e; }
